src/syntax/syntax.rkt

```diff
@@ -111,46 +154,100 @@
+    (and
+     (if (symbol? atypes)
+         (andmap (curry equal? atypes) actual-types)
+         (equal? atypes actual-types))
+     (cons true-name rtype))))
+
+(define (declare-parametric-operator! name op inputs output)
+  (hash-update! parametric-operators name (curry cons (list* op output inputs)) '())
+  (hash-set! parametric-operators-reverse op name))
+
+(define (operator-info* op field)
+  (cond
+   [(hash-has-key? parametric-operators op)
+    (let ([val (first (hash-ref parametric-operators op))])
+      (operator-info (car val) field))]
+   [else (operator-info op field)]))
+
+(define-operator (+.f64 binary64 binary64) binary64 
   [fl +] [bf bf+] [ival ival-add]
-  [->tex (curry format "~a + ~a")]
   [nonffi +])
 
-(define-operator (- real real) real
+(define-operator (-.f64 binary64 binary64) binary64
   [fl -] [bf bf-] [ival ival-sub]
-  [->tex (curry format "~a - ~a")]
   [nonffi -])
 
-(define-operator (neg real) real
+(define-operator (neg.f64 binary64) binary64
   [fl -] [bf bf-] [ival ival-neg]
-  [->tex (curry format "-~a")]
   [nonffi -])
 
-(define-operator (* real real) real
+(define-operator (*.f64 binary64 binary64) binary64
   [fl *] [bf bf*] [ival ival-mult]
-  [->tex (curry format "~a \\cdot ~a")]
   [nonffi *])
 
-(define-operator (/ real real) real
+(define-operator (/.f64 binary64 binary64) binary64
   [fl /] [bf bf/] [ival ival-div]
-  [->tex (curry format "\\frac{~a}{~a}")]
   [nonffi /])
 
+(declare-parametric-operator! '+ '+.f64 '(binary64 binary64) 'binary64)
+(declare-parametric-operator! '- '-.f64 '(binary64 binary64) 'binary64)
+(declare-parametric-operator! '- 'neg.f64 '(binary64) 'binary64)
+(declare-parametric-operator! '* '*.f64 '(binary64 binary64) 'binary64)
+(declare-parametric-operator! '/ '/.f64 '(binary64 binary64) 'binary64)
+
+(define-operator (+.f32 binary32 binary32) binary32 
+  [fl +] [bf bf+] [ival ival-add]
+  [nonffi +])
+
+(define-operator (-.f32 binary32 binary32) binary32
+  [fl -] [bf bf-] [ival ival-sub]
+  [nonffi -])
+
+(define-operator (neg.f32 binary32) binary32
+  [fl -] [bf bf-] [ival ival-neg]
+  [nonffi -])
+
+(define-operator (*.f32 binary32 binary32) binary32
+  [fl *] [bf bf*] [ival ival-mult]
+  [nonffi *])
+
+(define-operator (/.f32 binary32 binary32) binary32
+  [fl /] [bf bf/] [ival ival-div]
+  [nonffi /])
+
+(declare-parametric-operator! '+ '+.f32 '(binary32 binary32) 'binary32)
+(declare-parametric-operator! '- '-.f32 '(binary32 binary32) 'binary32)
+(declare-parametric-operator! '- 'neg.f32 '(binary32) 'binary32)
+(declare-parametric-operator! '* '*.f32 '(binary32 binary32) 'binary32)
+(declare-parametric-operator! '/ '/.f32 '(binary32 binary32) 'binary32)
+
+(define (declare-parametric-operator/libm op num-args)
+  (declare-parametric-operator! op (sym-append op '.f64) (make-list num-args 'binary64) 'binary64)
+  (declare-parametric-operator! op (sym-append op '.f32) (make-list num-args 'binary32) 'binary32))
+
```

Comment:
Generally find symbol manipulation like this ugly. Probably clearer to just write both names, especially since we'll be using both.